assets/js/modules/analytics/components/settings/PropertyOrWebDataStreamNotAvailableError.js

```diff
@@ -0,0 +1,140 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import { MODULES_ANALYTICS } from '../../datastore/constants';
+import { isValidPropertyID } from '../../../analytics-4/utils/validation';
+import ErrorText from '../../../../components/ErrorText';
+const { useSelect } = Data;
+
+export default function PropertyOrWebDataStreamNotAvailableError( props ) {
+	const { hasModuleAccess, isDisabled } = props;
+
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const measurementID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getMeasurementID()
+	);
+
+	const properties = useSelect( ( select ) =>
+		hasModuleAccess !== false && ! isDisabled
+			? select( MODULES_ANALYTICS_4 ).getProperties( accountID )
+			: []
+	);
+
+	const webDataStreams = useSelect( ( select ) =>
+		isValidPropertyID( propertyID )
+			? select(
+					MODULES_ANALYTICS_4
+			  ).getMatchingWebDataStreamsByPropertyID( propertyID )
+			: []
+	);
+
+	const getWebDataStreamsError = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector(
+			'getWebDataStreams',
+			[ propertyID ]
+		)
+	);
+
+	const getPropertiesError = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getProperties', [
+			accountID,
+		] )
+	);
+
+	if (
+		! hasModuleAccess ||
+		isDisabled ||
+		properties === undefined ||
+		webDataStreams === undefined ||
+		! isValidPropertyID( propertyID )
+	) {
+		return null;
+	}
+
+	const propertyAvailable = properties.some(
+		( { _id } ) => _id === propertyID
+	);
+
+	if (
+		propertyAvailable &&
+		measurementID &&
+		! getWebDataStreamsError &&
+		! webDataStreams.some(
+			( { webStreamData } ) =>
+				// eslint-disable-next-line sitekit/acronym-case
+				webStreamData.measurementId === measurementID
+		)
+	) {
+		return (
+			<ErrorText
+				message={ sprintf(
+					/* translators: 1: Google Analytics 4 Measurement ID. */
+					__(
+						'The previously selected web data stream with measurement ID %1$s is no longer available. Please select a new web data stream to continue collecting data with Google Analytics 4.',
+						'google-site-kit'
+					),
+					measurementID
+				) }
+			/>
+		);
+	}
+
+	if ( ! propertyAvailable && ! getPropertiesError ) {
```

Comment:
```suggestion
	if ( ! propertyAvailable && getPropertiesError === false ) {
```

Reply:
Ah, this is what was causing the bug I encountered: `getPropertiesError` can be `undefined` (for instance, while loading) and it causes a flag of this message when no property is available and there's no error.

Changing this to `=== false` fixes the issue.

Reply:
I was wrong about that, the issue is actually that the previously-selected property for an old account isn't reset until the new properties are loaded, meaning there's a time when the property ID is valid while things are still loading.

The trick should be "resetting" the property ID to be `undefined` when the account changes.